supervisor/dbus/agent/__init__.py

```diff
@@ -0,0 +1,99 @@
+"""OS-Agent implementation for DBUS."""
+import asyncio
+import logging
+from typing import Any, Dict
+
+from awesomeversion import AwesomeVersion
+
+from ...exceptions import DBusError, DBusInterfaceError
+from ...utils.gdbus import DBus
+from ..const import (
+    DBUS_ATTR_DIAGNOSTICS,
+    DBUS_ATTR_VERSION,
+    DBUS_NAME_HAOS,
+    DBUS_OBJECT_HAOS,
+)
+from ..interface import DBusInterface, dbus_property
+from ..utils import dbus_connected
+from .apparmor import AppArmor
+from .cgroup import CGroup
+from .datadisk import DataDisk
+from .system import System
+
+_LOGGER: logging.Logger = logging.getLogger(__name__)
+
+
+class OSAgent(DBusInterface):
+    """Handle D-Bus interface for OS-Agent."""
+
+    name = DBUS_NAME_HAOS
+
+    def __init__(self) -> None:
+        """Initialize Properties."""
+        self.properties: Dict[str, Any] = {}
+
+        self._cgroup: CGroup = CGroup()
+        self._apparmor: AppArmor = AppArmor()
+        self._system: System = System()
+        self._datadisk: DataDisk = DataDisk()
+
+    @property
+    def cgroup(self) -> CGroup:
+        """Return CGroup DBUS object."""
+        return self._cgroup
+
+    @property
+    def apparmor(self) -> AppArmor:
+        """Return AppArmor DBUS object."""
+        return self._apparmor
+
+    @property
+    def system(self) -> System:
+        """Return System DBUS object."""
+        return self._system
+
+    @property
+    def datadisk(self) -> DataDisk:
+        """Return DataDisk DBUS object."""
+        return self._datadisk
+
+    @property
+    @dbus_property
+    def version(self) -> AwesomeVersion:
+        """Return version of OS-Agent."""
+        return AwesomeVersion(self.properties[DBUS_ATTR_VERSION])
+
+    @property
+    @dbus_property
+    def diagnostics(self) -> bool:
+        """Return if diagnostics is enabled on OS-Agent."""
+        return self.properties[DBUS_ATTR_DIAGNOSTICS]
+
+    @diagnostics.setter
+    def diagnostics(self, value: bool) -> None:
+        """Enable oder disable OS-Agent diagnostics."""
+        asyncio.create_task(
+            self.dbus.set_property(DBUS_NAME_HAOS, DBUS_ATTR_DIAGNOSTICS, value)
+        )
+
+    async def connect(self) -> None:
+        """Connect to system's D-Bus."""
+        try:
+            self.dbus = await DBus.connect(DBUS_NAME_HAOS, DBUS_OBJECT_HAOS)
+            await self.cgroup.connect()
+            await self.apparmor.connect()
+            await self.system.connect()
+            await self.datadisk.connect()
```

Comment:
Should we have `self.diagnostics = self.sys_config.diagnostics`  as part of the connection here?
Or do you want to handle that elsewhere?

Reply:
right, will be later implemented on config + core setup process. For now, it just builds the bus interface